zeppelin-zengine/src/test/java/org/apache/zeppelin/resource/DistributedResourcePoolTest.java

```diff
@@ -44,8 +46,9 @@
   public void setUp() throws Exception {
     super.setUp();
     InterpreterSetting interpreterSetting = interpreterSettingManager.getByName("mock_resource_pool");
+    interpreterSetting.getOption().setPerNote(ISOLATED);
     intp1 = (RemoteInterpreter) interpreterSetting.getInterpreter("user1", "note1", "mock_resource_pool");
-    intp2 = (RemoteInterpreter) interpreterSetting.getInterpreter("user2", "note1", "mock_resource_pool");
+    intp2 = (RemoteInterpreter) interpreterSetting.getInterpreter("user2", "note2", "mock_resource_pool");
```

Comment:
While `intp1` and `intp2` were configured to the same Note `note1` with `SHARED` mode, the test was instantiating `intp1` and `intp2` in the same process and didn't test access to the remote object in the remote process.

Therefore, changed interpreter option to per note `ISOLATED` and configured `intp1` for `note1` and `intp2 for `note2`, make sure `intp1` and `intp2` runs in different process.

In this way, it can reproduce error described in ZEPPELIN-4004